core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -306,99 +306,48 @@ public static boolean isAuthenticationEnabled(AlluxioConfiguration conf) {
     return conf.getBoolean(PropertyKey.S3_REST_AUTHENTICATION_ENABLED);
   }
 
-  /**
-   * Convert header map key to lower case key.
-   * @param rawHeaders header map
-   * @return lower case key map
-   */
-  public static Map<LowerCaseKey, String> lowerCaseKeyMap(MultivaluedMap<String,
-                                                                String> rawHeaders) {
-    final Map<LowerCaseKey, String> headers = new HashMap<>();
-
-    for (Map.Entry<String, List<String>> headerEntry : rawHeaders.entrySet()) {
-      if (0 < headerEntry.getValue().size()) {
-        String headerKey = headerEntry.getKey();
-        if (!headers.containsKey(LowerCaseKey.valueOf(headerKey))) {
-          headers.put(new LowerCaseKey(headerKey), headerEntry.getValue().get(0));
-        }
-      }
-    }
-    return headers;
-  }
-
   /**
    * Convert MultivaluedMap to a single value map.
    *
-   * @param queryParameters
+   * @param queryParameters MultivaluedMap
+   * @param lowerCase whether to use lower case
    * @return a single value map
    */
   public static Map<String, String> fromMultiValueToSingleValueMap(
-      MultivaluedMap<String, String> queryParameters) {
-    Map<String, String> result = new HashMap<>();
+      MultivaluedMap<String, String> queryParameters, boolean lowerCase) {
+    Map<String, String> result = lowerCase
+        ? new TreeMap<>(new HeaderCaseComparator()) : new HashMap<>();
```

Comment:
You can define the comparator in lambda so you don't need to define a class for this: 
https://stackoverflow.com/questions/12947088/java-treemap-comparator

Reply:
finish.